base/src/main/groovy/org/asciidoctor/gradle/base/AbstractAsciidoctorBaseTask.groovy

```diff
@@ -92,7 +94,7 @@ abstract class AbstractAsciidoctorBaseTask extends DefaultTask {
      * @param f Any object convertible with {@code project.file}.
      */
     void setSourceDir(Object f) {
-        this.srcDir = f
+        this.srcDir.set(project.file(f))
```

Comment:
This evaluates the project early, which is exactly what needs to be avoided.

Reply:
Fair enough.  I've refactored it to use a provider to lazily evaluate whatever gets passed.  It's a little clunky because of the need to support the earlier Gradle versions which don't have great capabilities for what we're trying to do here, but it should work the way you want now.